components/lib/fetch-httpurlconnection/src/main/java/mozilla/components/lib/fetch/httpurlconnection/HttpURLConnectionClient.kt

```diff
@@ -32,6 +32,9 @@ class HttpURLConnectionClient : Client() {
 
     @Throws(IOException::class)
     override fun fetch(request: Request): Response {
+        if (request.private) {
+            throw IllegalStateException("Client doesn't support private request")
```

Comment:
nit: This should be `IllegalArgumentException`. `IllegalStateException` is used to indicate that the method can't be invoked.

Reply:
Generally it seem right to me to throw here, as otherwise we'd carry out the request on these clients without a notion of `private` which is likely not what the caller expected.